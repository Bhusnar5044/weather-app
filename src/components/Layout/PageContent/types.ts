import { PropsWithChildren } from 'react';

export interface Props extends PropsWithChildren {
    isLoading?: boolean;
}
