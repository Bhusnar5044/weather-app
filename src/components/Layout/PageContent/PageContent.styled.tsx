import { styled } from '@mui/material/styles';

export const PageContent = styled('main')`
    display: flex;
    flex-direction: column;
    gap: 2rem;
    overflow: auto;
    padding-top: 32px;
    width: 96%;
    margin: 0 auto;
    @media (min-width: 1536px) {
        max-width: 1536px;
    }
`;
