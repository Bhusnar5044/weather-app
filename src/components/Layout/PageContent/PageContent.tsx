import { FC, memo } from 'react';
import { Props } from './types';

export const PageContent: FC<Props> = memo(({ children }) => {
    return <main className="flex flex-col gap-2 w-full">{children}</main>;
});

PageContent.displayName = 'PageContent';
