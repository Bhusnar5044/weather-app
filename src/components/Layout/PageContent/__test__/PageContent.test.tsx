import { cleanup, render, screen } from '@test-utils';
import { PageContent } from '../PageContent';

describe('PageContent component', () => {
    beforeEach(() => {
        cleanup();
    });
    it('should render properly', () => {
        const { container } = render(<PageContent>Demo PageContent</PageContent>);
        expect(container).toMatchSnapshot();
    });

    it('should render with loader properly', () => {
        render(<PageContent isLoading>Demo PageContent</PageContent>);
        expect(screen.getByTitle('loader')).toBeInTheDocument();
        expect(screen.queryByText('Demo PageContent')).not.toBeInTheDocument();
    });
});
