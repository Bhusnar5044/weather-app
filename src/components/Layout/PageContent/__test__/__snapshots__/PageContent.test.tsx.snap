// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContent component should render properly 1`] = `
<div>
  <main
    class="flex flex-col gap-2 w-full"
  >
    Demo PageContent
  </main>
</div>
`;
